src/parser/lexer.mll

```diff
@@ -609,13 +611,8 @@ and code_span buffer nesting_level start_offset input = parse
     { Buffer.add_char buffer c;
       code_span buffer nesting_level start_offset input lexbuf }
 
-  | newline newline
-    { warning
-        input
-        (Parse_error.not_allowed
-          ~what:(Token.describe (`Blank_line "\n\n"))
-          ~in_what:(Token.describe (`Code_span "")));
-      Buffer.add_char buffer '\n';
+  | newline whitespace+
+    { Buffer.add_char buffer ' ';
```

Comment:
Something along the lines of (not tested):
```suggestion
  | newline horizontal_space (newline horizontal_space)+
    { warning
        input
        (Parse_error.not_allowed
          ~what:(Token.describe (`Blank_line "\n\n"))
          ~in_what:(Token.describe (`Code_span "")));
      Buffer.add_char buffer ' ';
      code_span buffer nesting_level start_offset input lexbuf }
  | newline horizontal_space+
    { Buffer.add_char buffer ' ';
```

Reply:
I've restored the warning, and fixed up the optional horizontal spaces, let me know what you think